File: folkmq/src/main/java/org/noear/folkmq/common/MqAssert.java
package org.noear.folkmq.common;

/**
 * 断言
 *
 * @author noear
 * @since 1.2
 */
public class MqAssert {
    /**
     * 断言元信息相关的名字与值
     * */
    public static void assertMeta(String str, String paramName) {
        if (str.indexOf('?') >= 0) {
            throw new IllegalArgumentException("Param '" + paramName + "' can't have symbols: '?'");
        }

        if (str.indexOf('&') >= 0) {
            throw new IllegalArgumentException("Param '" + paramName + "' can't have symbols: '&'");
        }

        if (str.indexOf('=') >= 0) {
            throw new IllegalArgumentException("Param '" + paramName + "' can't have symbols: '='");
        }

        if (str.indexOf('!') >= 0) {
            throw new IllegalArgumentException("Param '" + paramName + "' can't have symbols: '!'");
        }
    }
}

File: folkmq-lang-python/folkmq/folkmq/common/MqAssert.py
class MqAssert:
    @staticmethod
    def require_non_null(obj, message:str):
        if obj is None:
            raise Exception(message)
        else:
            return obj
    @staticmethod
    def assert_meta(str:str, paramName:str):
        MqAssert.assert_meta_symbols(str, paramName, '?', "?")
        MqAssert.assert_meta_symbols(str, paramName, '&', "&")
        MqAssert.assert_meta_symbols(str, paramName, '=', "=")
        MqAssert.assert_meta_symbols(str, paramName, '!', "!")

        MqAssert.assert_meta_symbols(str, paramName, '@', "@")
        MqAssert.assert_meta_symbols(str, paramName, '#', "#")
        MqAssert.assert_meta_symbols(str, paramName, '$', "$")
        MqAssert.assert_meta_symbols(str, paramName, '%', "%")
        MqAssert.assert_meta_symbols(str, paramName, '^', "^")
        MqAssert.assert_meta_symbols(str, paramName, '*', "*")

    @staticmethod
    def assert_meta_symbols(str: str, paramName: str, c:str, cS:str):
        if str.find(c) >= 0:
            raise Exception("Param '" + paramName + "' can't have symbols: '" + cS + "'");
